### ISSUE TITLE
Lexical editor seems to trigger autosave loop

### ISSUE THREAD

**Issue by taismassaro** [payloadcms/payload#3709]
### Link to reproduction

https://github.com/taismassaro/payload/blob/main/test/_community/collections/Posts/index.ts

### To Reproduce

- Add Lexical editor to a collection that has `autosave` enabled
- In the admin UI, open the Network tab in the DevTools and open an item from the collection
- Notice that the file immediately starts autosaving (calls keep getting triggered and the `Last saved` message keeps getting updated)

### Describe the Bug

Something in how the Lexical editor works keeps updating the field value and triggering the autosave functionality in a loop. This doesn't happen in other fields or with the Slate editor, which only autosaves on blur.

### Payload Version

2.0.7

### Adapters and Plugins
```
"@payloadcms/bundler-webpack": "^1.0.3", 
"@payloadcms/db-mongodb": "^1.0.3",
"@payloadcms/plugin-cloud": "^2.2.5",
"@payloadcms/richtext-lexical": "^0.1.12",
```

**Comment by AlessioGr** [comment]
Hey @taismassaro 

I cannot reproduce the issue, see the video:

https://github.com/payloadcms/payload/assets/70709113/38c8eae7-6d34-435a-b935-7ee8c41233ae

It only saves when data in the editor has been changed. Maybe this has already been fixed! Could you test if you can replicate this in [v0.1.13](https://www.npmjs.com/package/@payloadcms/richtext-lexical/v/0.1.13) of @payloadcms/richtext-lexical?

**Comment by taismassaro** [comment]
hey @AlessioGr, thanks for checking. I'm wondering if the problem we have is that the rich text field is part of an array?
also I see in your video that you immediately clicked the publish button after typing in the editor and I see the saving loop right after typing, so can you just double check that? I know my description said it would immediately trigger but with the newest version it doesn't seem to, but still happens after typing.

https://github.com/payloadcms/payload/assets/16339834/1a23809d-1877-44bf-88fd-9f10c56e09a4

**Comment by revnelson** [comment]
Seeing the same issue on my end. Latest versions of payload and lexical at the time of this posting. Fields config looks like:

```
const fields: Field[] = [
	{ name: "title", type: "text", required: true },
	{
		type: "tabs",
		tabs: [
			{
				label: "Description",
				fields: [
					{
						name: "description",
						label: false,
						type: "richText",
						editor: lexicalEditor({
							features({ defaultFeatures }) {
								return [...defaultFeatures]
							}
						})
					}
				]
			},
                ]
        }
]

**Comment by craigrdaniels** [comment]
I am getting the same autosave loop in lexical if I include a block in rich text content.

eg:

```export const RichTextContent: Block = {
  slug: "richTextContent",
  labels: {
    singular: "Rich Text Content",
    plural: "Rich Text Content Blocks",
  },
  fields: [
    {
      name: "content",
      label: "Content",
      type: "richText",
      required: true,
      editor: lexicalEditor({
        features: ({ defaultFeatures }) => [
          ...defaultFeatures,
          BlocksFeature({
            blocks: [CTA],
          }),
        ],
      }),
    },
  ],
};
```
If I use the lexical editor without the block the loop won't trigger.

```      editor: lexicalEditor({
        features: ({ defaultFeatures }) => [
          ...defaultFeatures,
        ],
      }),
```

Just adding the block outside of the rich text editor will also not trigger the loop.

**Comment by AlessioGr** [comment]
Fixed by 9277b306de6300dcda838398e72090ee0669c5d9

Looks like passing `props` into the dependencies array of the validate callback caused infinite re-rendering - but only after the draft was first saved. I don't really know why the props were changing (specifically, props.admin), but removing it from the dependencies array fixed it
